api/python/t4/packages.py

```diff
@@ -391,6 +391,21 @@ def walk(self):
                 for key, value in child.walk():
                     yield name + '/' + key, value
 
+    def _walk_dir_meta(self):
+        """
+        Generator that traverses all entries in the package tree and returns
+            tuples of (key, meta) for each directory with metadata.
+        Keys will all end in '/' to indicate that they are directories.
+        """
+        for key, child in sorted(self._children.items()):
+            if isinstance(child, PackageEntry):
+                continue
+            meta = child.get_meta()
+            if meta:
+                yield key + '/', meta
+            for key, meta in child._walk_dir_meta():
+                yield key, meta
```

Comment:
Doesn't the original `key` get lost here?

Reply:
I'm not sure what you mean. I do think it's a good idea to not shadow `key` so I'll use a different name for the child keys, but I'm not sure that's what you're asking